libsolidity/ast/Types.h

```diff
@@ -165,8 +165,8 @@ class Type
 {
 public:
 	Type() = default;
-	Type(Type const&) = delete;
-	Type(Type&&) = delete;
+	Type(Type const&) = default;
```

Comment:
Why do we disallow AST types to be copy constructible?

Reply:
I think in my use case right now, I could also only use the move-constructor. Is it ok to allow this class to be move constructible?